plugin/pkg/admission/podtolerationrestriction/admission.go

```diff
@@ -162,6 +164,21 @@ func (p *podTolerationsPlugin) Admit(a admission.Attributes) error {
 		}
 	}
 
+	if qoshelper.GetPodQOS(pod) != api.PodQOSBestEffort {
```

Comment:
we need to ensure that this admission controller is run after all resource defaulting controllers.